### PR TITLE
Add stm32l0 support for ExtiPin futures

### PR DESCRIPTION
**Pull request by michaelbeaumont**
Made possible with #48, tested to work on my `stm32l072`. It's quite similar to `embassy-stm32f4` so far, with some changes because of differences between `stm32l0xx-hal` and `stm32f4xx-hal`.

### CODE REVIEW

**Comment by thalesfragoso** [comment]
This seems to also hit #55

**Comment by michaelbeaumont** [comment]
Yes I used the same steps as with `f4` at that point. 

Good point about the `static mut`. I added that because of the way `exti::listen_gpio` works. It may end up needing to be split up in the HAL so that some config can happen in `new_pin` (probably similar to how it's done with `f4`).

**Comment by Dirbaio** [comment]
If the issue is it needs to do RMW of registers to configure the pins, I'd be OK with the exti driver taking small critical sections to do that. That should allow creating multiple pins and probably fix #55 too (?)

**Comment by thalesfragoso** [comment]
> If the issue is it needs to do RMW of registers to configure the pins, I'd be OK with the exti driver taking small critical sections to do that.

In the F4 we can probably use bit-banding, not sure if the L0 has bit-banding though.

**Comment by michaelbeaumont** [comment]
@Dirbaio @thalesfragoso Is the last commit what you had in mind?

File: embassy-stm32l0/src/exti.rs [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+    {
+        ExtiPin {
+            pin,
+            interrupt,
+            mgr: self,
+        }
+    }
+}
+
+pub struct ExtiPin<T, I> {
```
Comment (xoviat) [inline embassy-stm32l0/src/exti.rs:39]: Remove the `I` type parameter by adding type bounds to the `Interrupt` type. See stm32f4 for an example.

**Review by xoviat — APPROVED** [review]

**Pull request merged**